account_invoice_triple_discount/models/line_triple_discount.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class LineTripleDiscount (models.AbstractModel):
+    _name = "line.triple_discount"
+    _description = "Add triple discount fields and logic"
+
+    discount = fields.Float()
+    discount2 = fields.Float(
+        'Discount 2 (%)',
+        digits=dp.get_precision('Discount'),
+    )
+    discount3 = fields.Float(
+        'Discount 3 (%)',
+        digits=dp.get_precision('Discount'),
+    )
+    discounting_type = fields.Selection(
```

Comment:
I don't understand where this field can be changed ? Should not be a setting on the related ``res.partner`` ? I mean some suppliers are using ``additive`` method and other ``multiplicative`` method.

Don't you think ?

Reply:
This can be changed directly on the account invoice line's form, but it can be interesting to inherit that from the partner.
I'm currently investigating why Travis is still 🔴 (that's why this is still a Draft), after that I can try to add this new behavior.

Reply:
While implementing this feature I noticed that it is not as trivial as it seems because default value for invoice line cannot rely on the line's fields (default methods are `api.model`), so I'd need to save it in the context etc..
I think that deserves its own PR, and here let's focus on the described issue, is that ok for you?

BTW Now that Travis is 🟢, I'll mark this PR as ready for reviews

Reply:
well, I was more thinking on storing the discounting_type only on the ``res.partner`` or at least on the order/invoice method.
it will be a lighter implementation.
Do you think there are real cases when there are some line with ``additive`` method, and other with ``multiplicative`` ?

I mean, setting this value 50 times, if you have 50 lines will be a mess, in a UX point of view.

Reply:
That is why I was talking about a default value for new lines: each of the 50 lines would have the default value inherited from the partner; and switching partner in the invoice would trigger the edit of all the invoice's lines.

I don't know about real cases of lines having different kinds of discounts but this is how sale orders + triple_discount currently work and it would be nice to have it in invoices too.

Reply:
I added the logic explained above, can you update your review?

Reply:
> I don't know about real cases of lines having different kinds of discounts but this is how sale orders + triple_discount currently work and it would be nice to have it in invoices too.

1) Well if it's not a use case, it should be changed I think. Don't you ? 


2)the problem with the current implementation is that it will create inconsistencies. I mean, it's possible to have "multiplicative" and invoice level and "additive" on line level.

3) you are adding new field on invoice / invoice.line. don't you think you should add a pre-migration script to populate correctly the database. I fear that an update of the current module on a production database will be slow. Don't you think ?